### PR TITLE
Added support for causal SDPA

### PR DESCRIPTION
**Pull request by mikkelfo**
Rather than forcing non-causal SDPA, we can just remove the `attn_mask` (SDPA doesn't allow `causal=True` and `attn_mask` to be set). 

Alternatively, we can have an if/else clause in the `forward_sdpa` that sets `attn_mask = None` instead of doing it in `mha`, but I don't believe the user (we) care for this.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved causal attention handling for more reliable sequence processing.
  * Causal mode now correctly manages attention masks.
  * Invalid attention configurations continue to provide clear validation messages.

### CODE REVIEW

File: bonsai/modules/networks/bonsai_nets.py [hunk 1]
```diff
@@ -80,11 +80,11 @@ def forward(self, batch):
 
         if self.hparams["attn_type"] == "flash":
             return self.forward_flash(x, batch["attention_mask"])
-        elif self.hparams["attn_type"] == "sdpa" and not self.hparams["causal"]:
+        elif self.hparams["attn_type"] == "sdpa":
             return self.forward_sdpa(x, batch["attention_mask"])
```
Comment (Sllambias) [inline bonsai/modules/networks/bonsai_nets.py]: could be handled by `attn_mask=batch["attention_mask"] if not self.hparams["causal"] else None`

**Review by Sllambias — APPROVED** [review]
Tiny suggested change to make it a little bit simpler and intuitive otherwise lgtm
